core/tools/blueos_startup_update/blueos_startup_update.py

```diff
@@ -438,6 +438,20 @@ def ensure_user_data_structure_is_in_place() -> bool:
     return False
 
 
+def build_led_overlay():
+    overlay_exists = locate_file(["/boot/overlays/spi0-led.dtbo", "/boot/firmware/overlays/spi0-led.dtbo"])
+    if overlay_exists:
+        logger.info(f"spi0-led overlay found at {overlay_exists}")
+        return False
+    with open("/install/spi0-led.dts", "r", encoding="utf-8") as f:
+        dts = f.read()
+        save_file("/tmp/spi0-led.dts", dts, "")
+    command = "sudo dtc -@ -Hepapr -I dts -O dtb -o /boot/overlays/spi0-led.dtbo /tmp/spi0-led.dts && sudo cp /boot/overlays/spi0-led.dtbo /boot/firmware/overlays/spi0-led.dtbo"
```

Comment:
Just to be sure, both paths exist in both versions ? (bullseye and bookworm)
Shouldn't this be done based in the existence of the path ?

Reply:
good catch. it does't exist on bullseye. I added a check